src/main/java/org/jenkinsci/plugins/github_branch_source/PullRequestSCMHead.java

```diff
@@ -327,9 +328,10 @@ public FixMetadataMigration() {
         @Override
         public PullRequestSCMHead migrate(@NonNull GitHubSCMSource source, @NonNull FixMetadata head) {
             PullRequestSource src = source.retrievePullRequestSource(head.getNumber());
+            GitHubSCMSourceHelper helper = GitHubSCMSourceHelper.build(source);
             return new PullRequestSCMHead(head.getName(), src == null ? null : src.getSourceOwner(),
                     src == null ? null : src.getSourceRepo(), src == null ? null : src.getSourceBranch(),
-                    head.getNumber(), head.getTarget(), src != null && source.getRepoOwner().equalsIgnoreCase(src.getSourceOwner())
+                    head.getNumber(), head.getTarget(), src != null && helper.owner.equalsIgnoreCase(src.getSourceOwner())
```

Comment:
The fact that you have to change existing calls to the getters to use the helper to be able to handle the possibilities of null/empty values is a bit worrisome to me. Why can't the getters in `GitHubSCMSource` compute their return values using something like the methods in the helper so that their nullability is unchanged and these other calls don't have to be modified? 

We already have `boolean isURLRepo()` being used to know which version of the UI to show to users, so it seems like the overall approach would work.

Reply:
I thought about it before knowing about the outdoors complexity, probably now makes more sense.